src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,87 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+    private User user;
+
+    @BeforeEach
+     void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        user = new User();
+    }
+
+    @Test
+    void userIsNull_notOk() {
+        assertThrows(CustomException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void userLogin_notOk() {
+        user.setLogin("login1414");
+        user.setPassword("password1414");
+        user.setAge(25);
+        storageDao.add(user);
```

Comment:
Using `storageDao.add(user)` directly in tests is not recommended. Instead, you should use low-level methods to insert test data into storage. This ensures that the tests are not dependent on the correctness of the `register` method, which is what you are testing.